semana-10/projeto-labex/labex/src/screens/Admin/TripDetailsPage.js

```diff
@@ -0,0 +1,148 @@
+import React, {useEffect} from 'react'
+import axios from 'axios'
+import Header from '../../components/Header'
+
+//routes
+import { useHistory, useParams } from "react-router-dom";
+import {goBack, goToLoginPage} from '../../router/goToPages'
+
+// hook:
+import useRequestData from '../../hooks/useRequestData';
+
+// api:
+import {baseUrl, student} from '../../requisitions/apiRequisitions.js'
+
+// images:
+import profile from '../../img/profile.svg'
+import decideTrue from '../../img/decideTrue.svg'
+import decideFalse from '../../img/decideFalse.svg'
+import loading from '../../img/loading.gif'
+
+// styles:
+import {Subtitle, TripInfoTitle, TripInfo, BackLink, Loading} from '../../styles/MainStyles'
+import { CarouselTitle, CarouselDetails } from '../../styles/Carousel'
+
+import {AdminPageContainer, AdminPanelContainer} from './styled'
+import {CandidatesGrid, CandidateContainer, ProfileImg, DecideContainer, DecideButton, DecideImg, DecideLabel} from './styled'
+
+
+const TripDetailsPage = () => {
+
+    // - - - - ROUTE - - - -
+
+    const history = useHistory()
+    const pathParams = useParams()
+
+    // - - - - API - - - -
+
+    const getTripDetailUrl = `${baseUrl}${student}/trip/${pathParams.id}`
+    const auth = localStorage.getItem("token")
+
+    // - - - - ESTADOS & CANDIDATOS - - - -
+    
+    const [tripsDetails, setTripsDetails] = useRequestData(getTripDetailUrl, {}, auth)
+
+    const trip = tripsDetails.trip
+    const candidates = tripsDetails.trip && tripsDetails.trip.candidates
+
+    useEffect( () => {
+        const token = window.localStorage.getItem("token")
+
+        if (!token) {
+            goToLoginPage(history)
+        }
+
+    }, [history] )
+
+
+    const decideCandidate = (candidateId, boolean) => {
+
+        const decideCandidateUrl = `${baseUrl}${student}/trips/${pathParams.id}/candidates/${candidateId}/decide`
+
+        const decideCandidateBody = {
+            approve: boolean,
+        }
+
+        axios
+        .put(decideCandidateUrl, decideCandidateBody, {
+            headers: {
+                auth: window.localStorage.getItem("token")
+            }
+        })
+        .then( (response) => {
+            if (boolean) {
+                alert("Candidato aprovado com sucesso!")
+            } else {
+                alert("Candidato recusado")
+            }
+            window.location.reload(true);
```

Comment:
Achei massa, não conhecia essa função! O problema dela é que atualiza a página... Em desenvolvimento isso vai funcionar bem, mas em produção (tu pode testar isso lá no surge) quando a página atualiza ele vai dar page not found... Isso acontece porque a rota daquela página só quem conhece é o JavaScript, apesar da rota estar lá no navegador. Quando a gnt atualiza ou busca uma rota especifica da aplicação ela vai dar page not found, porque ela vai buscar um servidor com aquela rota, mas esse servidor não existe, uma vez que as rotas são gerenciadas pelo React. 
É um pouco confuso isso, acho que não fui muito claro, mas é algo até que tinha anotado pra perguntar pros professores como resolver isso